src/test/java/de/rwth/idsg/steve/web/validation/ChargeBoxIdValidatorTest.java

```diff
@@ -0,0 +1,151 @@
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package de.rwth.idsg.steve.web.validation;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+/**
+ * @author Sevket Goekay <sevketgokay@gmail.com>
+ * @since 01.08.2024
+ */
+public class ChargeBoxIdValidatorTest {
+
+    ChargeBoxIdValidator validator = new ChargeBoxIdValidator();
+
+    @Test
+    public void testNull() {
+        Assertions.assertTrue(validator.isValid(null, null));
+    }
+
+    @Test
+    public void testAllLowercaseLetters() {
+        Assertions.assertTrue(validator.isValid("test", null));
+    }
+
+    @Test
+    public void testAllUppercaseLetters() {
+        Assertions.assertTrue(validator.isValid("TEST", null));
+    }
+
+    @Test
+    public void testMixedCaseLetters() {
+        Assertions.assertTrue(validator.isValid("TesT", null));
+        Assertions.assertTrue(validator.isValid("tEst", null));
+    }
+
+    @Test
+    public void testLettersAndNumbers() {
+        Assertions.assertTrue(validator.isValid("test12", null));
+        Assertions.assertTrue(validator.isValid("89test", null));
+        Assertions.assertTrue(validator.isValid("te9s0t", null));
+    }
+
+    @Test
+    public void testDot() {
+        Assertions.assertTrue(validator.isValid(".test", null));
+        Assertions.assertTrue(validator.isValid("test.", null));
+        Assertions.assertTrue(validator.isValid("te..st", null));
+    }
+
+    @Test
+    public void testDash() {
+        Assertions.assertTrue(validator.isValid("-test", null));
+        Assertions.assertTrue(validator.isValid("test-", null));
+        Assertions.assertTrue(validator.isValid("te--st", null));
+    }
+
+    @Test
+    public void testUnderscore() {
+        Assertions.assertTrue(validator.isValid("_test", null));
+        Assertions.assertTrue(validator.isValid("test_", null));
+        Assertions.assertTrue(validator.isValid("te__st", null));
+    }
+
+    @Test
+    public void testColon() {
+        Assertions.assertTrue(validator.isValid(":test", null));
+        Assertions.assertTrue(validator.isValid("test:", null));
+        Assertions.assertTrue(validator.isValid("te::st", null));
+        Assertions.assertTrue(validator.isValid("VID:00XXXXXXXXXX", null));
+    }
+
+    @Test
+    public void testPoundSign() {
+        Assertions.assertTrue(validator.isValid("#test", null));
+        Assertions.assertTrue(validator.isValid("test#", null));
+        Assertions.assertTrue(validator.isValid("te##st", null));
+        Assertions.assertTrue(validator.isValid("#FreeCharging", null));
+    }
+
+    @Test
+    public void testCombined() {
+        Assertions.assertTrue(validator.isValid("1t.E-S_:t20#", null));
+    }
+
+    @Test
+    public void testEmpty() {
+        Assertions.assertFalse(validator.isValid("", null));
+    }
+
+    @Test
+    public void testSpace() {
+        Assertions.assertFalse(validator.isValid("  ", null));
```

Comment:
You should test some other blank, not printable, characters from the ascii table which are currently accepted.